File: PW/ConfigEditor/src/Structures/ConfigList.java
package Structures;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by io on 13.10.14.
 */
public class ConfigList
{
    public String _name;
    public List<Type> _types = new ArrayList<Type>();

    public ConfigList(String name,String values,String types)
    {
        _name = name;
        String[] vals = values.split(";");
        String[] tty = types.split(";");
        for(int i = 0; i < vals.length;i++)
            _types.add(new Type(vals[i],tty[i]));
    }
    public int getSize()
    {
        int result = 0;
        for(Type t : this._types)
            result += t._size;
        return result;
    }
}

File: PW/pwAPI/pwAPI/StructuresElement/ConfigList.cs
using System.Collections.Generic;
using System.IO;

namespace pwApi.StructuresElement
{
    public class ConfigList
    {
        public string Name;
        public string Skip;
        public Type[] Types;

        public ConfigList(string name, string skip, string values, string types)
        {
            Name = name;
            Skip = skip;
            var vals = values.Split(';');
            var tty = types.Split(';');
            Types = new Type[vals.Length];
            for(var i = 0; i < vals.Length; i++)
                Types[i] = new Type(vals[i], tty[i]);
        }
    }

    public class ConfigLists
    {
        public byte NpcTalk;
        public List<ConfigList> Lists;

        public void ParseList(string path)
        {
            Lists = new List<ConfigList>();
            var args = File.ReadAllLines(path);
            var count = int.Parse(args[0]);
            NpcTalk = byte.Parse(args[1]);
            var line = 2;
            for(var i = 0; i < count; i++)
            {
                while(args[line] == "")
                    line++;
                Lists.Add(new ConfigList(args[line], args[line + 1], args[line + 2], args[line + 3]));
                line += 4;
            }
        }
    }
}
